src/lib.rs

```diff
@@ -25,23 +26,27 @@ fn check_mut_ptr<T>(ptr: *mut T) -> Result<*mut T> {
     }
 }
 
-fn check_retcode(n: &Nfs, code: i32) -> Result<()> {
+fn check_retcode(ctx: *mut nfs_context, code: i32) -> Result<()> {
     if code < 0 {
-        Err(Error::new(ErrorKind::Other, n.get_nfs_error()?))
+        unsafe {
+            let err_str = nfs_get_error(ctx);
+            let e = CStr::from_ptr(err_str).to_string_lossy().into_owned();
+            Err(Error::new(ErrorKind::Other, e))
+        }
     } else {
         Ok(())
     }
 }
 
 pub struct Nfs {
-    context: *mut nfs_context,
+    context: Rc<*mut nfs_context>,
```

Comment:
This doesn't make much sense - a raw pointer doesn't own anything, so managing one in Rc<> doesn't really help. I'd expect the type in here to be something that owns the underlying object and implements Drop so that it gets cleaned up.

I'd suggest something like:
```
struct NfsPtr(*mut nfs_context);
impl Drop for NfsPtr {
  fn drop(&mut self) {
     unsafe { nfs_destroy_context(self.0) }
  }
}

struct Nfs(Rc<NfsPtr>);
```

Reply:
Yeah my lack of knowledge around how Rc works is showing.  I pushed another commit with your feedback.